app/web/src/store/views.store.ts

```diff
@@ -1559,7 +1559,7 @@ export const useViewsStore = (forceChangeSetId?: ChangeSetId) => {
             )
               this.syncUrlIntoSelection();
             const viewId = router.currentRoute.value.params.viewId;
-            if (viewId) {
+            if (viewId && this.selectedViewId !== viewId) {
```

Comment:
ah yeah as soon as you mentioned the watcher I realized that we didn't add a check for if you are already in that view! 🤪

Reply:
🧠